hal/src/nRF52840/spi_hal.cpp

```diff
@@ -672,12 +672,8 @@ int32_t hal_spi_acquire(hal_spi_interface_t spi, const hal_spi_acquire_config_t*
     if (!hal_interrupt_is_isr()) {
         os_mutex_recursive_t mutex = spiMap[spi].mutex;
         if (mutex) {
-            // FIXME: os_mutex_recursive_lock doesn't take any arguments, using trylock for now
-            if (!conf || conf->timeout != 0) {
```

Comment:
This still needs to be supported. If `conf && conf->timeout == 0` we need to trylock due to wiring `SPIClass::trylock()`.

Reply:
Same for Gen4?

Reply:
Yup.

Reply:
But again, only if config is provided and timeout is 0.